### PR TITLE
Improved edns0 options handling

### PR DESCRIPTION
**Pull request by UladzimirTrehubenka**
ATM we have support for splitting any edns0 option to some separate attributes,
syntax:  edns0 [code]  [name] [dataType] [destType] [start] [end]
where [dataType] = hex, [start] is the start index in byte array, [end] is the end index in byte array + 1
Example:
edns0 0xffee customer_id hex string 0 16
edns0 0xffee client_id hex string 16 32
Attributes (data has size 32 bytes - getting from the option with 0xffee code):
customer_id = data[0:16] (first 16 bytes)
client_id = data[16:32] (last 16 bytes)

### CODE REVIEW

File: contrib/coredns/policy/policy.go [hunk 1]
```diff
@@ -245,7 +250,7 @@ func (p *PolicyPlugin) handlePermit(ctx context.Context, w dns.ResponseWriter,
 		if debugQuery {
 			return p.retDebugInfo(r, w, respDomain, nil, "failed")
 		}
-		return status, err
+		return p.retRcode(w, r, dns.RcodeServerFailure, err)
```
Comment (rdrozhdzh) [inline contrib/coredns/policy/policy.go]: Why did you change status to dns.RcodeServerFailure?

Comment (UladzimirTrehubenka) [reply contrib/coredns/policy/policy.go]: If err != nil status is RcodeServerFailure, so the logic is not broken.

Comment (UladzimirTrehubenka) [reply contrib/coredns/policy/policy.go]: status is used later in the code

File: contrib/coredns/policy/policy.go [hunk 2]
```diff
@@ -171,29 +171,34 @@ func (p *PolicyPlugin) getAttrsFromEDNS0(r *dns.Msg, ip string) []*pb.Attribute
...
-			value = ip.String()
-		}
-		from := option.stringOffset
-		to := option.stringOffset + option.stringSize
-		if to > 0 && to <= len(value) && from < to {
-			value = value[from:to]
-		}
-		if option.name == "source_ip" {
-			ipId = "proxy_source_ip"
+		for _, option := range options {
```
Comment (vasili-v) [inline contrib/coredns/policy/policy.go]: May be extract this to separate function to put following here:
```go
...
group, srcIp := parseOptionGroup(options)
attrs = append(attrs, group...)
if srcIp {
	ipId = "proxy_source_ip"
}
...
```
